src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue

```diff
@@ -168,16 +183,33 @@
             ...mapGettersByNames(namespace, ["data", "filters", "countryAreaFilterOption"]),
             ...mapGetters("metadata", ["sapIndicatorsMetadata"]),
             ...mapGetters("plottingSelections", ["selectedSAPColourScales"]),
+            currentLanguage: mapStateProp<RootState, Language>(
+                null,
+                (state: RootState) => state.language
+            ),
             filterTableIndicators() {
                 return this.sapIndicatorsMetadata.filter((val: ChoroplethIndicatorMetadata) => val.indicator === this.plottingSelections.indicatorId)
+            },
+            dataSourceOptions() {
+                const options = [];
+                const lang = {lng: this.currentLanguage};
+                if (this.survey.available) {
+                    options.push({id: "2", label: i18next.t("survey", lang)});
```

Comment:
I guess you could pass in the translation key rather than the translation itself, and then set up the `tree-select` `option-label` to translate it, but I see any advantage besides reducing a tiny bit of repetition.

Reply:
Good idea

Reply:
Actually, I am just going to leave that is - worked ok to put the translate into the option and value slots of the treeselect, but that ended up being just as verbose as this, it's very nasty to unit test, and I've ended up with race condition failures I can't face chasing down right now!

Reply:
Sorry - one of those ideas that's only good in theory!